### ISSUE TITLE
Client Instantiation can be simplified

### ISSUE THREAD

**Issue by myuwono** [Covata/delta-sdk-python#76]
There's a couple of ways this can be done:

1. Via indirect getter: I'm not a fan of this approach personally; but since the API Client have a public `key_store` property exposed, we can pass in an instantiated `ApiClient` object and let the Client get its key_store instance from the api client. i.e.

```python
class Client:
    def __init__(self, api_client):
        """
        Creates a new DeltaClient instance from the provided configuration.

        :param api_client: an :class:`~.ApiClient` object
        :type api_client: class:`~.ApiClient`
        """
        self.__api_client = api_client
        self.__key_store = api_client.key_store
```

2. Via functional composition: I like this better. We can treat classes in python as yet another function, so we can "compose" `key_store` into any function that manufactures an `ApiClient`. Here goes: Pass in a `key_store` object and a `api_client_factory(key_store): ApiClient ` method (`ApiClient.__init__` qualifies this criterion btw). The `key_store` object will then be passed into the factory method in `Client.__init__`.

```python
from covata.delta import FileSystemKeyStore, ApiClient


class Client:
    def __init__(self, key_store, api_client_factory):
        """
        Creates a new DeltaClient instance from the provided configuration.

        :param key_store: the key store
        :type key_store: :class:`~.DeltaKeyStore`
        :param api_client_factory: the API client factory
        :type api_client_factory:
           (:class:`~.DeltaKeyStore`) -> :class:`~.ApiClient`
        """
        self.__key_store = key_store
        self.__api_client = api_client_factory(key_store)

def main():
    key_store = FileSystemKeyStore("~/keystore/", "passPhrase")
    client = Client(key_store, api_client_factory=ApiClient)
```

Which direction should we take?

**Comment by numbat** [comment]
I like approach 2.

We should default to our implementation of ApiClient so we have a simplified initialisation:

```python
    key_store = FileSystemKeyStore("~/keystore/", "passPhrase")
    client = Client(key_store)
```

**Comment by myuwono** [comment]
closed via #81
